content/shmem_malloc.tex

```diff
@@ -47,7 +47,11 @@
     are provided  so that multiple \ac{PE}s in a program can allocate symmetric,
     remotely accessible memory blocks.  These memory blocks can then be used with
     \openshmem communication routines.  Each of these routines call the
-    \FUNC{shmem\_barrier\_all} routine before returning; this ensures that all
+    \FUNC{shmem\_barrier\_all} routine before returning: \FUNC{shmem\_malloc} calls a
```

Comment:
What do you think of replacing:
> Each of these routines call the `shmem_barrier_all` routine before returning: [...]

with:

> Each of these routines include at least one call to `shmem_barrier_all`: [...]

The former, specifically the statement "before returning", could be misinterpreted as "_immediately_ before returning", even though you later clarify when the barrier(s) happen. I think this minor change would remove any potential ambiguity or confusion.